### PR TITLE
add multi download for portal

### PR DESCRIPTION
**Pull request by patrickweh**
minor fixes in ProductDetail.php

### CODE REVIEW

File: routes/frontend/portal.php [hunk 1]
```diff
@@ -88,6 +91,21 @@
 
             Route::get('/media/{media}/{filename}', function (Media $media) {
                 return $media;
-            });
+            })->name('media');
+
+            Route::any('/media/download-multiple', function (Request $request) {
```
Comment (SirSplasch) [inline routes/frontend/portal.php:96]: why any instead of get?

Comment (SirSplasch) [inline routes/frontend/portal.php:96]: Why use callback and not run it through a controller?

File: src/Livewire/Portal/Shop/ProductDetail.php [hunk 2]
```diff
@@ -56,13 +75,38 @@ public function selectOption(ProductOption $option): void
         }
 
         $this->fillProductForm($product);
+        $this->variant = $product->id;
+    }
+
+    public function downloadMedia(array|int $media, ?string $collectionName = null): void
+    {
+        $media = app(MediaModel::class)->query()->whereIntegerInRaw('id', Arr::wrap($media))->get();
```
Comment (SirSplasch) [inline src/Livewire/Portal/Shop/ProductDetail.php:83]: wrap this query

File: src/Livewire/Portal/Shop/ProductDetail.php [hunk 3]
```diff
@@ -20,11 +24,26 @@ class ProductDetail extends Component
...
         $this->fillProductForm($product);
+
+        if ($this->variant) {
+            $variant = app(Product::class)
+                ->query()
+                ->where('parent_id', $product->id)
+                ->whereKey($this->variant)
+                ->with('productOptions:id,product_option_group_id')
+                ->sole();
+            $this->groups = $variant->productOptions->pluck('id', 'product_option_group_id')->toArray();
```
Comment (SirSplasch) [inline src/Livewire/Portal/Shop/ProductDetail.php:43]: this does not work as intended.
The array is set up with ['product_option_group_id' => 'id'], and therefore if multiple product options have the same product_option_group_id only the last one is shown.

File: src/Livewire/Portal/Shop/ProductDetail.php [hunk 4]
```diff
@@ -113,11 +165,58 @@ protected function fillProductForm(Product $product): void
...
+            'addGlobalScope',
+            [
+                'scope' => 'bundleProducts',
+                'implementation' => fn (Builder $query) => $query->with([
+                    'bundleProducts' => fn ($query) => $query->withPivot('count'), 'media',
+                ]),
+            ]
+        );
+
+        $this->getBundleData($product->bundleProducts()->get());
```
Comment (SirSplasch) [inline src/Livewire/Portal/Shop/ProductDetail.php:196]: eager load bundleProducts media?

File: src/Livewire/Portal/Shop/ProductDetail.php [hunk 5]
```diff
@@ -113,11 +165,58 @@ protected function fillProductForm(Product $product): void
...
+                /** @var Media $media */
+                $this->productForm->additionalMedia[$media->collection_name][$media->id] = [
+                    'id' => $media->id,
+                    'name' => $media->name,
+                    'url' => $media->getUrl(),
+                ];
+            }
+
+            if ($product->bundleProducts->isNotEmpty()) {
+                $this->getBundleData($product->bundleProducts);
```
Comment (SirSplasch) [inline src/Livewire/Portal/Shop/ProductDetail.php:218]: eager load bundleProducts media?

File: src/Livewire/Portal/Shop/ProductDetail.php [hunk 6]
```diff
@@ -56,13 +75,38 @@ public function selectOption(ProductOption $option): void
...
+    public function downloadMedia(array|int $media, ?string $collectionName = null): void
+    {
+        $media = app(MediaModel::class)->query()->whereIntegerInRaw('id', Arr::wrap($media))->get();
+
+        count($media) > 1
+            ? $this->redirectRoute(
+                'portal.media.download-multiple',
+                [
+                    'ids' => $media->pluck('id')->implode(','),
+                    'filename' => $this->productForm->name . ' - ' . $collectionName,
```
Comment (SirSplasch) [inline src/Livewire/Portal/Shop/ProductDetail.php:90]: only append collectionName if not null, otherwise the filename ends with ' - '

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: routes/frontend/portal.php (hunk 1 above)
Comment (patrickweh) [reply routes/frontend/portal.php:96]: √

File: src/Livewire/Portal/Shop/ProductDetail.php (hunk 2 above)
Comment (patrickweh) [reply src/Livewire/Portal/Shop/ProductDetail.php:83]: √

File: src/Livewire/Portal/Shop/ProductDetail.php (hunk 5 above)
Comment (patrickweh) [reply src/Livewire/Portal/Shop/ProductDetail.php:218]: loaded via global scope

File: src/Livewire/Portal/Shop/ProductDetail.php (hunk 6 above)
Comment (patrickweh) [reply src/Livewire/Portal/Shop/ProductDetail.php:90]: √

**Pull request merged**